### PR TITLE
Rough stab at making the default template settings-driven

### PR DESCRIPTION
**Pull request by davidfells**
This is a quick stab at converting the templating to be dynamic. I was not able to get the project to build locally, so this is completely untested, but the changes appear to be quite simple. Along with other users in the issue thread at https://github.com/lynchjames/obsidian-day-planner/issues/56 , being able to customize my default Day Planner would be a fantastic QOL improvement.

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export default class DayPlannerFile {
         try {
             const normalizedFileName = normalizePath(fileName);
             if (!await this.vault.adapter.exists(normalizedFileName, false)) {
-                await this.vault.create(normalizedFileName, DAY_PLANNER_DEFAULT_CONTENT);
+                await this.vault.create(normalizedFileName, DayPlannerSettings.defaultContent);
```
Comment (Thracky) [inline src/file.ts:64]: Needs to be `this.settings.defaultContent`

File: src/planner-md.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export default class PlannerMarkdown {
         const fileContents = await (await this.file.getFileContents(filePath)).split('\n');
         const view = this.workspace.activeLeaf.view as MarkdownView;
         const currentLine = view.sourceMode.cmEditor.getCursor().line;
-        const insertResult = [...fileContents.slice(0, currentLine), ...DAY_PLANNER_DEFAULT_CONTENT.split('\n'), ...fileContents.slice(currentLine)];
+        const insertResult = [...fileContents.slice(0, currentLine), ...DayPlannerSettings.defaultContent.split('\n'), ...fileContents.slice(currentLine)];
```
Comment (Thracky) [inline src/planner-md.ts:35]: Also needs to be `this.settings.defaultContent`

**Review by Thracky — COMMENTED** [review]
I hugely appreciate this addition as I felt it was the only major thing missing from the plugin and wanted to give it a whirl so I fixed up the build (failing due to API changes) and was able to test and fix a couple things so I'll outline them all here.

Not directly related to this PR but to get it to build I did the following:
in [tsconfig.json](https://github.com/lynchjames/obsidian-day-planner/blob/d1eb7ce187e7757b7a3880358a6ee184b3b025da/tsconfig.json) :
remove `"inlineSourceMap": true,"

in [main.ts](https://github.com/lynchjames/obsidian-day-planner/blob/d1eb7ce187e7757b7a3880358a6ee184b3b025da/src/main.ts):

On line 46 change:
```typescript
    this.registerEvent(this.app.on("codemirror", this.codeMirror));
```
to
```typescript 
    this.registerCodeMirror(this.codeMirror);
```
The API was changed in v0.10.0 in https://github.com/obsidianmd/obsidian-api/commit/2e4ac7c8dd4d1b8ca4cf726eac75a935a3ea25d8#diff-0eaea5db2513fdc5fe65d534d3591db5b577fe376925187c8a624124632b7466

Also, I feel like it would be nice to specify a template file instead of having to jam markdown into the tiny little config text box. The core Daily notes plugin does this but not sure how specifically. However, I'm pretty happy with the setting for now :)

**Comment by balint-bg** [comment]
Just wanted to say that I'm eagerly waiting for this feature to be merged. I can't comment on the code (I know nothing about JS) but in case a huge +1 was going to help getting this in sooner, then here it is :)
Thanks for the work!!!

**Comment by lynchjames** [comment]
> I hugely appreciate this addition as I felt it was the only major thing missing from the plugin and wanted to give it a whirl so I fixed up the build (failing due to API changes) and was able to test and fix a couple things so I'll outline them all here.
> 
> Not directly related to this PR but to get it to build I did the following:
> in [tsconfig.json](https://github.com/lynchjames/obsidian-day-planner/blob/d1eb7ce187e7757b7a3880358a6ee184b3b025da/tsconfig.json) :
> remove `"inlineSourceMap": true,"
> 
> in [main.ts](https://github.com/lynchjames/obsidian-day-planner/blob/d1eb7ce187e7757b7a3880358a6ee184b3b025da/src/main.ts):
> 
> On line 46 change:
> 
> ```ts
>     this.registerEvent(this.app.on("codemirror", this.codeMirror));
> ```
> 
> to
> 
> ```ts
>     this.registerCodeMirror(this.codeMirror);
> ```
> 
> The API was changed in v0.10.0 in [obsidianmd/obsidian-api@2e4ac7c#diff-0eaea5db2513fdc5fe65d534d3591db5b577fe376925187c8a624124632b7466](https://github.com/obsidianmd/obsidian-api/commit/2e4ac7c8dd4d1b8ca4cf726eac75a935a3ea25d8#diff-0eaea5db2513fdc5fe65d534d3591db5b577fe376925187c8a624124632b7466)
> 
> Also, I feel like it would be nice to specify a template file instead of having to jam markdown into the tiny little config text box. The core Daily notes plugin does this but not sure how specifically. However, I'm pretty happy with the setting for now :)

@Thracky the inlineSourceMap setting has now been removed from tsconfig.json.

I decided to replace the CodeMirror event with a file modified event as the Day Planner does not need to be updated on every single change in the CodeMirror editor, modifications to the file are enough to keep the gantt chart, status bar and timeline updated.

**Comment by lynchjames** [comment]
@davidfells could you rebase from lynchjames:main to get the latest updates and make the changes @Thracky has suggested?

**Comment by zdway10** [comment]
hello, I want to load specific markdown template. it's great to know you done it. And I had downloaded those 4 changed files with updates. Question: How can i enable it ? need I update the main.js ? thanks in advanve.

**Comment by ebdavison** [comment]
Would love to see this merged as well as this would be a big benefit over hacking the JS code to change the daily template.

**Comment by gkiar** [comment]
:+1: would also love this feature

**Comment by djdaniels90** [comment]
I am going to take this up, i made the necessary changes here and have it working locally.  However I am going to add extra functionality around templater.  If templater is installed and you provide a path to the template file then it will parse the template through templater first.

**Comment by takirami** [comment]
Also following this feature closely. Seems like it would add some great benefits

**Comment by ivan-lednev** [comment]
Closing this as stale. If you think this is still valuable, please rewrite the implementation (the code in `main` has been rewritten) and open another PR.

**Pull request closed without merging**